File: codeforces/C_Move_Brackets.cpp
/*
    https://codeforces.com/problemset/problem/1374/C
    Time: O(n)
    Space: O(1)
*/

#include <iostream>

using namespace std;

int main()
{
    int t;
    cin >> t;

    while (t--)
    {
        int n;
        cin >> n;

        string s;
        cin >> s;

        int ans = 0, ps = 0;
        for (auto c : s)
        {
            if (c == '(')
                ps++;
            else
                ps--;

            if (ps < 0)
            {
                ans++;
                ps = 0;
            }
        }

        cout << ans << endl;
    }
}

File: codeforces/C_Move_Brackets.py

# https://codeforces.com/problemset/problem/1374/C
# Time: O(N)
# Space: O(N)

for _ in range(int(input())):

    n, s = int(input()), input()

    cnt = res = 0

    for x in s:
        cnt += 1 if x == '(' else -1
        if cnt < 0:
            res += 1
            cnt += 1

    print(res)
